packages/core/data-transfer/src/strapi/providers/remote-destination/index.ts

```diff
@@ -205,13 +224,13 @@ class RemoteStrapiDestinationProvider implements IDestinationProvider {
     const wsUrl = `${wsProtocol}//${url.host}${url.pathname}${TRANSFER_PATH}`;
     // No auth defined, trying public access for transfer
     if (!auth) {
-      ws = new WebSocket(wsUrl);
+      ws = await this.#connectToWebsocket(wsUrl);
     }
 
     // Common token auth, this should be the main auth method
     else if (auth.type === 'token') {
       const headers = { Authorization: `Bearer ${auth.token}` };
-      ws = new WebSocket(wsUrl, { headers });
+      ws = await this.#connectToWebsocket(wsUrl, headers, auth);
```

Comment:
Should we just have `#connectToWebsocket` handle this logic? Since we're already passing in auth to get the token out of it, it might be better off to let it construct the necessary headers as well

Reply:
I am not sure, we can of load the whole logic of websocket to this method too but am still not sure if this is something we want to do, I would like to treat this method as just a wrapper for `new Websocket()` that makes it async so we can await it.